### DISCUSSION TITLE
INAV Radar Wiring with a T3_V1.6.1 Board

### DISCUSSION THREAD

**Discussion by jongell** [iNavFlight/inav#10374] [Q&A] [answered]
Hello! 

This is my first time using INAV Radar and getting ESP32 boards. I successfully flashed both boards, they are running LilyGo16_433Mhz according to https://www.dropbox.com/scl/fo/18lnbuealvjxe7wvdvdub/AJXdaTW92tht7MZ0f5LCMaE?e=1&file_subpath=%2FESP32+Radar&preview=ESP32+Radar.302.zip&rlkey=0t7xi9e34itnmjvthnik4224k

Now I'm confused about wiring. The old modules have Pin 17/23 for TX/RX

My boards are missing those pins and in RCGroups it says Pin 1 and 3 for Board v1.6 (https://www.rcgroups.com/forums/showthread.php?3304673-ESP32-Radar-for-INAV-(position-sharing-between-aircrafts)) but also those are missing. 

Any idea where to go from here? Could these modules I bought even work? (Left: Modules I bought, Right: Old Modules with Pin 17/23)

<img width="1440" alt="Screenshot 2024-09-21 at 15 32 09" src="https://github.com/user-attachments/assets/dd2cdbfc-d6ef-4eda-98af-176dd38e5b46">

**Comment by mmosca** [comment]
you may want to switch to: https://github.com/FormationFlight/FormationFlight
as a bonus, you can use it with elrs receivers that are smaller

**Reply by jongell** [reply]
Looks like the recommended ESP_Flasher for the formation flight flashing is down:

Using '/dev/cu.usbserial-0001' as serial port.
Connecting....
Detecting chip type... ESP32
Connecting...

Chip Info:
 - Chip Family: ESP32
 - Chip Model: ESP32-PICO-D4 (revision v1.1)
 - Number of Cores: 2
 - Max CPU Frequency: 240MHz
 - Has Bluetooth: YES
 - Has Embedded Flash: YES
 - Has Factory-Calibrated ADC: YES
 - MAC Address: E8:6B:EA:1A:9C:44
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 1500000
Changed.
 - Flash Size: 4MB
Unexpected error: Error while retrieving firmware file 'https://raw.githubusercontent.com/Jason2866/ESP_Flasher/factory/bootloader/esp32/bin/bootloader_dout_40m.elf': 404 Client Error: Not Found for url: https://raw.githubusercontent.com/Jason2866/ESP_Flasher/factory/bootloader/esp32/bin/bootloader_dout_40m.elf

No idea where to look next

**Reply by jongell** [reply, marked as answer]
Turns out it was all easier than anticipated (shocker) :D

Just wire the T3_v1.6.1 TXD and RXD to a free UART (TX to RX and RX to TX), and enable MSP with Baud 115200 on that UART.
Then, do `get hud` in cli and configure the HUD OSD in INAV. Apart from enabling crosshairs there is No need to turn on radar or anything else in OSD, just HUD related parameters. 

For me I also needed a MAG as heading sensor for the Radar characters to appear in the OSD. Not sure why since 7.1 should support magless dead reckoning. Either way, It works very good with crazy range.

I even added a small hack such that I can communicate between 2 vehicles over the Lora Modules. As soon as the GCS NAV Mode is active on one vehicle it sets the gpsSol.groundCourse response of the MSP_RAW_GPS message to -3210 which can be parsed on the other vehicle as a custom action. Just check if the radar_poi[nearest_poi].heading is -321. This also works very well with minimal delay of around 1s
